### PR TITLE
Feature/rework login with angularjs

### PR DESCRIPTION
**Pull request by anatoliiibrahimov**
#[Link to JIRA ticket](https://ssu-jira.softserveinc.com/browse/LR-192)

### Author.

- [x] I have added link to this PR to the JIRA
- [x] I have logged time in JIRA.

### Reviewers.

- [x] @YUzhva 
- [x] @OKrain 
- [x] @vovka

### Issue summary.

*Reworked login with angular*

### Fix summary. 

_Log in with popup_

### Testing approach.

Rspec tests

### CODE REVIEW

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 1]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
     $scope.quiz = Quiz.show({quizId: $routeParams.quizId});
   }
 ]);
+
+quizzesControllers.controller('SettingsCtrl', ['$scope', '$http',
+function($scope, $http) {   
+   $scope.handlePopupAuthentication = function handlePopupAuthentication(network, account) {
+      $scope.$apply(function(){
+        $scope.applyNetwork(network, account);
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: What is `applyNetwork`?

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -5,15 +5,20 @@ def facebook
     @user = User.from_omniauth(request.env["omniauth.auth"])
 
     if @user.persisted?
-      sign_in_and_redirect @user, event: :authentication
+      sign_in @user, event: :authentication
       # this will throw if @user is not activated
       set_flash_message(:notice, :success, kind: "Facebook") if is_navigational_format?
+      redirect_to logged_in_success_path
```
Comment (vovka) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Why did you separate `sign_in` and `redirect`?

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -5,15 +5,20 @@ def facebook
...
     else
       session["devise.facebook_data"] = request.env["omniauth.auth"]
       redirect_to root_path
     end
   end
 
+  def success
+    render "home/success"
```
Comment (vovka) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Please, follow conventions. Move `home/success` file to `users/omniauth_callbacks` folder. And then remove explicit `render` from action.

File: app/views/home/index.html.erb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<div class="container">
+<div class="container" ng-controller="SettingsCtrl">
```
Comment (vovka) [inline app/views/home/index.html.erb]: Please, use router to bind controller to page.

File: app/views/home/success.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<p>This view will now self-destruct</p>
+<script>
+   try {
+      window.opener.$windowScope.handlePopupAuthentication('<%= @provider %>', <%= @account.to_json.html_safe %>);
```
Comment (vovka) [inline app/views/home/success.html.erb]: What is `@provider`? What is `@account`?

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -15,7 +15,7 @@
   </head>
   <body>
     <header class="navbar navbar-default navbar-fixed-top">
-      <div class="container">
+      <div class="container" ng-controller="SettingsCtrl">
```
Comment (vovka) [inline app/views/layouts/application.html.erb]: Please, use router to bind controller to page.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 7]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+      });
+    };
+
+   $scope.authNetwork = function authNetwork(network) {
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Looks like `network` argument is not used.

File: config/initializers/devise.rb [hunk 8]
```diff
@@ -1,6 +1,7 @@
 Devise.setup do |config|
   config.omniauth :facebook, Rails.application.secrets.facebook_app_id,
-                             Rails.application.secrets.facebook_app_secret
+                             Rails.application.secrets.facebook_app_secret,
+                   display:  'popup'
```
Comment (vovka) [inline config/initializers/devise.rb]: Is it required? As I understand, you wrote own popup open logic.

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 2 above)
Comment (vovka) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Why redirect to success page after user already signed in?

File: app/views/home/success.html.erb [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+   } catch(err) {}
+   window.close();
+   window.opener.location.reload();
```
Comment (vovka) [inline app/views/home/success.html.erb]: Is this native page reload, without angular?

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 10]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
     $scope.quiz = Quiz.show({quizId: $routeParams.quizId});
   }
 ]);
+
+quizzesControllers.controller('SettingsCtrl', ['$scope', '$http',
```
Comment (YUzhva) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Is there a better name for this controller to describe what it actually do?

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 11]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+function($scope, $http) {   
+   $scope.handlePopupAuthentication = function handlePopupAuthentication(network, account) {
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 2 space characters but found 3.](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l88)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 7 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ["network" is defined but never used](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l94)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 12]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+      $scope.$apply(function(){
+        $scope.applyNetwork(network, account);
+      });
+    };
+
+   $scope.authNetwork = function authNetwork(network) {
+      var openUrl = 'users/auth/facebook';
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l95)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 13]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
   }
 ]);
+
+quizzesControllers.controller('SettingsCtrl', ['$scope', '$http',
+function($scope, $http) {   
+   $scope.handlePopupAuthentication = function handlePopupAuthentication(network, account) {
+      $scope.$apply(function(){
+        $scope.applyNetwork(network, account);
+      });
+    };
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 2.](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l92)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 7 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 2 space characters but found 3.](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l94)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 14]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
     $scope.quiz = Quiz.show({quizId: $routeParams.quizId});
   }
 ]);
+
+quizzesControllers.controller('SettingsCtrl', ['$scope', '$http',
+function($scope, $http) {   
+   $scope.handlePopupAuthentication = function handlePopupAuthentication(network, account) {
+      $scope.$apply(function(){
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l89)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 15]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+        $scope.applyNetwork(network, account);
+      });
+    };
+
+   $scope.authNetwork = function authNetwork(network) {
+      var openUrl = 'users/auth/facebook';
+      window.$windowScope = $scope;
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l96)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 16]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+      window.open(openUrl, "Authenticate Account", "width=500, height=500");
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l97)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 17]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
     $scope.quiz = Quiz.show({quizId: $routeParams.quizId});
   }
 ]);
+
+quizzesControllers.controller('SettingsCtrl', ['$scope', '$http',
+function($scope, $http) {
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ["$http" is defined but never used](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l87)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 18]
```diff
@@ -82,3 +82,19 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+        $scope.applyNetwork(network, account);
+      });
+    };
+
+   $scope.authNetwork = function authNetwork(network) {
+      var openUrl = 'users/auth/facebook';
+      window.$windowScope = $scope;
+      window.open(openUrl, "Authenticate Account", "width=500, height=500");
+    };
+  }
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 0 space characters but found 2.](https://www.codacy.com/app/vovka/video_quiz/file/2572984367/issues/source?bid=3205144&fileBranchId=3298567#l99)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 19]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
     $scope.quiz = Quiz.show({quizId: $routeParams.quizId});
   }
 ]);
+
+quizzesControllers.controller('popUpCtrl', ['$scope', '$http',
+function($scope, $http) {   
+   $scope.handlePopupAuthentication = function handlePopupAuthentication(account) {
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ['account' is defined but never used.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l88)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 20]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
     $scope.quiz = Quiz.show({quizId: $routeParams.quizId});
   }
 ]);
+
+quizzesControllers.controller('popUpCtrl', ['$scope', '$http',
+function($scope, $http) {
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ['$http' is defined but never used.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l87)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 19 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 2 space characters but found 3.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l88)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 21]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+quizzesControllers.controller('popUpCtrl', ['$scope', '$http',
+function($scope, $http) {   
+   $scope.handlePopupAuthentication = function handlePopupAuthentication(account) {
+      $scope.$apply(function(){
+      });
+    };
+
+   $scope.authNetwork = function authNetwork() {
+      var openUrl = 'users/auth/facebook';
+      window.$windowScope = $scope;
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l95)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 22]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+      window.open(openUrl, "Authenticate Account", "width=500, height=500");
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l96)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 23]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
 ]);
+
+quizzesControllers.controller('popUpCtrl', ['$scope', '$http',
+function($scope, $http) {   
+   $scope.handlePopupAuthentication = function handlePopupAuthentication(account) {
+      $scope.$apply(function(){
+      });
+    };
+
+   $scope.authNetwork = function authNetwork() {
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 2 space characters but found 3.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l93)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 24]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+      var openUrl = 'users/auth/facebook';
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l94)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 25]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
     $scope.quiz = Quiz.show({quizId: $routeParams.quizId});
   }
 ]);
+
+quizzesControllers.controller('popUpCtrl', ['$scope', '$http',
+function($scope, $http) {   
+   $scope.handlePopupAuthentication = function handlePopupAuthentication(account) {
+      $scope.$apply(function(){
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l89)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 26]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+      });
+    };
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 2.](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l91)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 19 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ["account" is defined but never used](https://www.codacy.com/app/vovka/video_quiz/file/2576023456/issues/source?bid=3205144&fileBranchId=3298567#l88)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 10 above)
Comment (anatoliiibrahimov) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: renamed

File: config/initializers/devise.rb (hunk 8 above)
Comment (anatoliiibrahimov) [inline config/initializers/devise.rb]: yes,it is required for interface like popup login

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 3 above)
Comment (anatoliiibrahimov) [inline app/controllers/users/omniauth_callbacks_controller.rb]: done

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 19 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ['account' is defined but never used.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l88)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 20 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ['$http' is defined but never used.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l87)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 19 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 2 space characters but found 3.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l88)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 21 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l95)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 22 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l96)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 23 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 2 space characters but found 3.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l93)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 24 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l94)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 25 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 5 space characters but found 6.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l89)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 26 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 2.](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l91)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 19 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ["account" is defined but never used](https://www.codacy.com/app/vovka/video_quiz/file/2576071543/issues/source?bid=3205144&fileBranchId=3298567#l88)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 27]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+  $scope.handlePopupAuthentication = function handlePopupAuthentication() {
+    $scope.$apply(function(){
+    });
+  };
+
+  $scope.authNetwork = function authNetwork() {
+    var openUrl = 'users/auth/facebook';
+    window.$windowScope = $scope;
+    window.open(openUrl, "Authenticate Account", "width=500, height=500");
+    };
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 2.](https://www.codacy.com/app/vovka/video_quiz/file/2576104300/issues/source?bid=3205144&fileBranchId=3298567#l97)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 20 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ["$http" is defined but never used](https://www.codacy.com/app/vovka/video_quiz/file/2576104300/issues/source?bid=3205144&fileBranchId=3298567#l87)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 28]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
...
+  }
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 0 space characters but found 2.](https://www.codacy.com/app/vovka/video_quiz/file/2576104300/issues/source?bid=3205144&fileBranchId=3298567#l98)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 20 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ['$http' is defined but never used.](https://www.codacy.com/app/vovka/video_quiz/file/2576104300/issues/source?bid=3205144&fileBranchId=3298567#l87)

Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ['$http' is defined but never used.](https://www.codacy.com/app/vovka/video_quiz/file/2577482349/issues/source?bid=3205144&fileBranchId=3298567#l87)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 27 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 2.](https://www.codacy.com/app/vovka/video_quiz/file/2577482349/issues/source?bid=3205144&fileBranchId=3298567#l97)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 20 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: ["$http" is defined but never used](https://www.codacy.com/app/vovka/video_quiz/file/2577482349/issues/source?bid=3205144&fileBranchId=3298567#l87)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 28 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Expected indentation of 0 space characters but found 2.](https://www.codacy.com/app/vovka/video_quiz/file/2577482349/issues/source?bid=3205144&fileBranchId=3298567#l98)

File: config/initializers/devise.rb [hunk 29]
```diff
@@ -1,6 +1,7 @@
 Devise.setup do |config|
   config.omniauth :facebook, Rails.application.secrets.facebook_app_id,
-                             Rails.application.secrets.facebook_app_secret
+                             Rails.application.secrets.facebook_app_secret,
+                   display: 'popup'
```
Comment (vovka) [inline config/initializers/devise.rb]: Codacy Issue found: [Align the parameters of a method call if they span more than one line.](https://www.codacy.com/app/vovka/video_quiz/file/2577482313/issues/source?bid=3205144&fileBranchId=3298567#l4)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 30]
```diff
@@ -82,3 +82,18 @@ quizzesControllers.controller('showQuizCtrl', ['$scope', 'Quiz', '$routeParams',
     $scope.quiz = Quiz.show({quizId: $routeParams.quizId});
   }
 ]);
+
+quizzesControllers.controller('popUpCtrl', ['$scope', '$http',
+function($scope, $http) {   
+  $scope.handlePopupAuthentication = function handlePopupAuthentication() {
+    $scope.$apply(function(){
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: What does this method do?

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 31]
```diff
@@ -2,18 +2,21 @@ class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
...
-    @user = User.from_omniauth(request.env["omniauth.auth"])
-
+    @user = User.from_omniauth(request.env["omniauth.auth"])    
     if @user.persisted?
-      sign_in_and_redirect @user, event: :authentication
+      sign_in @user, event: :authentication
+      redirect_to logged_in_success_path
       # this will throw if @user is not activated
-      set_flash_message(:notice, :success, kind: "Facebook") if is_navigational_format?
+      set_flash_message(:notice, :success, kind: "Facebook") if is_navigational_format?
```
Comment (vovka) [inline app/controllers/users/omniauth_callbacks_controller.rb]: trailing spaces

**Comment by vovka** [comment]
:ok:

**Pull request merged**